### DISCUSSION TITLE
Custom hook/hook options for react-query code generator

### DISCUSSION THREAD

**Discussion by raff-r** [dotansimha/graphql-code-generator#6455] [Q&A]
Before using graphql-codegen we used a custom hook to wrap around `useQuery` hook from react-query which included default options such as `staleTime`, `enabled` etc.

Is it possible to use a custom hook when generating hooks or include predefined query options for each generated hook to contain?

**Comment by dotansimha** [comment]
You can customize some aspects of the generated code (like, the hook used), but note that the signature must be the same. 
You can always create a custom plugin that will generate output based on your specific needs :)
